cluster/saltbase/salt/kube-master-addons/kube-master-addons.sh

```diff
@@ -45,15 +45,24 @@ while true; do
     elif [[ $rc == 124 ]]; then
       restart_docker=true
     fi
+
+    timeout 30 docker load -i /srv/salt/kube-bins/federated-apiserver.tar 1>/dev/null 2>&1
```

Comment:
You should use a deployment, which creates a replicaset for you.

On Thu, May 5, 2016 at 8:47 AM, Quinton Hoole <notifications@github.com>
wrote:

> In cluster/saltbase/salt/kube-master-addons/kube-master-addons.sh
> <https://github.com/kubernetes/kubernetes/pull/25125#discussion_r62206608>
> :
>
> > @@ -45,15 +45,24 @@ while true; do
> >      elif [[ $rc == 124 ]]; then
> >        restart_docker=true
> >      fi
> > +
> > +    timeout 30 docker load -i /srv/salt/kube-bins/federated-apiserver.tar 1>/dev/null 2>&1
>
> I would prefer us to create ReplicaSet to run the Federated API server.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/25125/files/e80616bed4bafaf5f31fc26db9040d843946809d#r62206608>
>